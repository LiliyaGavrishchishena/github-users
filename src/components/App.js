import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import queryString from 'query-string';
import API from '../services/api';
import SearchInput from './SearchInput/SearchInput';
import ReposList from './ReposList/ReposList';

class App extends Component {
  state = { username: '', repos: [] };

  componentDidMount() {
    const { location } = this.props;
    const { username } = this.state;

    const parsed = queryString.parse(location.search);

    if (parsed.username && parsed.username !== username) {
      this.setState({ username: parsed.username });
    }
    API.getReposOfUser(username).then(
      data => Array.isArray(data) && this.setState({ repos: data }),
    );
  }

  handleChangeInput = e => {
    const { value } = e.target;
    this.setState({ username: value });
  };

  handleSubmit = e => {
    e.preventDefault();
    const { history, location } = this.props;
    const { username } = this.state;

    const usernameString = queryString.stringify({ username });

    history.push({
      pathname: location.pathname,
      search: usernameString,
    });

    API.getReposOfUser(username).then(
      data => Array.isArray(data) && this.setState({ repos: data }),
    );
  };

  render() {
    const { repos, username } = this.state;
    return (
      <div>
        <SearchInput
          username={username}
          onChange={this.handleChangeInput}
          onSubmit={this.handleSubmit}
        />
        {repos.length > 0 && <ReposList repos={repos} />}
      </div>
    );
  }
}

export default withRouter(App);
