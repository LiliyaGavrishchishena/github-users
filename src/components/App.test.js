import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import App from './App';

describe('App', () => {
  const wrapper = shallow(<App />);
  it('renders without crashing', () => {
    console.log(wrapper.debug());
  });
  it('matches the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
