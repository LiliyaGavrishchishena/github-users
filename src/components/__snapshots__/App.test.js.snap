// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`First React component test with Enzyme matches the snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
